core/src/torii/mod.rs

```diff
@@ -240,92 +223,80 @@ impl<W: WorldTrait> Torii<W> {
                 })
             });
 
-        let router = warp::post()
-            .and(post_router)
+        let router = ws_router
+            .or(warp::post().and(post_router))
             .or(warp::get().and(get_router))
-            .or(ws_router)
             .with(warp::trace::request())
             .recover(Torii::<W>::recover_arg_parse);
 
         tokio::spawn(async move {
-            start_status(Arc::clone(&state))
+            self.start_status()
                 .await
                 .wrap_err("Failed to start status service")
                 .unwrap_or_else(|error| {
                     iroha_logger::error!(%error);
                 })
         });
 
-        match self.iroha_cfg.torii.api_url.to_socket_addrs() {
+        match api_url.to_socket_addrs() {
             Ok(mut i) => {
                 #[allow(clippy::expect_used)]
                 let addr = i.next().expect("ToSocketAddrs iteration failed");
                 warp::serve(router).run(addr).await;
                 Ok(())
             }
             Err(error) => {
-                iroha_logger::error!(%self.iroha_cfg.torii.api_url, %error, "API address configuration parse error");
+                iroha_logger::error!(%api_url, %error, "API address configuration parse error");
                 Err(eyre::Error::new(error))
             }
         }
     }
-}
 
-/// Start status endpoint.
-///
-/// # Errors
-/// Can fail due to listening to network or if http server fails
-async fn start_status<W: WorldTrait>(state: ToriiState<W>) -> eyre::Result<()> {
-    let get_router_status = endpoint1(
-        handle_status,
-        warp::path(uri::STATUS).and(add_state(Arc::clone(&state))),
-    );
-    let get_router_metrics = endpoint1(
-        handle_metrics,
-        warp::path(uri::METRICS).and(add_state(Arc::clone(&state))),
-    );
-    let router = warp::get().and(get_router_status).or(get_router_metrics);
-
-    match state.iroha_cfg.torii.status_url.to_socket_addrs() {
-        Ok(mut i) => {
-            #[allow(clippy::expect_used)]
-            let addr = i.next().expect("ToSocketAddrs iteration failed");
-            warp::serve(router).run(addr).await;
-            Ok(())
-        }
-        Err(e) => {
-            iroha_logger::error!("Status address configuration parse error");
-            Err(eyre::Error::new(e))
+    /// Start status endpoint.
+    ///
+    /// # Errors
+    /// Can fail due to listening to network or if http server fails
+    async fn start_status(self) -> eyre::Result<()> {
```

Comment:
We only need `self.wsv` and `self.network` here, maybe it is best not to move `self` into func? Just take them as arguments.

I think there is a high probability that we might need to start services on other ports.

Reply:
mhm, I thought this was like a chain of calls to start services. One service is started and then forwards `self` to another to start itself and so on

Reply:
But as the task is spawned, basically we move torii inside this spawned task and consume it, the way it is currently implemented.

Reply:
When it comes to further services on different ports in the future,
I prefer existing(flat) spawns to chained(nested) spawns for readability and modularity

Reply:
I'm going to use flat spawns, will commit changes soon